snapshot/lib/snapshot/simulator_launchers/simulator_launcher.rb

```diff
@@ -89,7 +89,7 @@ def launch_simultaneously(devices, language, locale, launch_arguments)
         log_path: xcodebuild_log_path(language: language, locale: locale)
       )
 
-      devices.each { |device_type| override_status_bar(device_type) } if launcher_config.override_status_bar
+      devices.each { |device_type| override_status_bar(device_type, launcher_config.override_status_bar_arguments) } if launcher_config.override_status_bar
```

Comment:
Should this "if" be based on the option with arguments (the new one)?
If not, do we need the line above this one?

Reply:
No, the `if` is based on whether or not `launcher_config.override_status_bar` is `true`. If it's not, it doesn't matter what the arguments are. If it is, it'll use the arguments if they're set, or a default if not. So it should always work, and be backwards compatible for anyone that hasn't included specific arguments.

Reply:
But if no arguments are provided (I.e. `nil`) the if gets evaluated to `false` and then `override_status_bar` won't be called, so no simulator will be launched and no default arguments will be set in the status bar or I'm missing something there?

Reply:
The `if` is evaluating `launcher_config.override_status_bar`, not `launcher_config.override_status_bar_arguments` right? So it shouldn't matter whether `launcher_config.override_status_bar_arguments` is set to a string or if it's `nil`. There are two separate configuration options:

```ruby
override_status_bar(true)
override_status_bar_arguments("--time 9:41 --dataNetwork lte --cellularMode active --cellularBars 4 --batteryState charging --batteryLevel 75 --operatorName TELUS")
```

As long as `override_status_bar` is `true`, `devices.each` should iterate through, and pass `launcher_config.override_status_bar_arguments` to the `override_status_bar` method, whether it's set to a string (valid or empty), or `nil`. If it's empty or `nil`, the updated `override_status_bar` method will set it to the same arguments that were used in the previous version to maintain backwards compatibility.

That being said, this is my first time writing ruby code, so I could be completely wrong. ;)

Reply:
I misread the if condition, you're totally right

Reply:
🤦 sorry for this! May be because I reviewed this on iOS app but I thought you were doing 2 subsequent calls:

```ruby
devices.each { |device_type| override_status_bar(device_type) } if launcher_config.override_status_bar
devices.each { |device_type| override_status_bar(device_type, launcher_config.override_status_bar_arguments) } if launcher_config.override_status_bar
```

That's why I mentioned the 1st one shouldn't be needed 😅 

All good, your logic is 💯💯